scanomatic/image_analysis/grid_cell_extra.py

```diff
@@ -602,7 +602,7 @@ def detect(
             if self.filter_array.sum() == 0:
                 self.filter_array = self.old_filter.copy()
 
-            blob_diff = (np.abs(self.old_filter - self.filter_array)).sum()
+            blob_diff = (np.abs(self.old_filter ^ self.filter_array)).sum()
```

Comment:
can you explain this and the following changes, why they are valid.

Reply:
`self.old_filter` and `self.filter_array` are numpy boolean arrays. Doing the subtraction results in the following error
```
TypeError: numpy boolean subtract, the `-` operator, is not supported, use the bitwise_xor, the `^` operator, or the logical_xor function instead.
```
I assumed that we wanted the same behaviour as subtracting regular booleans which is
```
True - True = 0
True - False = 1
False - False = 0
False - True = -1
```
The `^` operator on np arrays does
```
np.array(True) ^ np.array(True) = False
np.array(True) ^ np.array(False) = True
np.array(False) ^ np.array(False) = False
np.array(False) ^ np.array(True) = True
```
which should do the same thing in combination with the `np.abs` and `.sum()`

Reply:
but then you don't need the `abs` any more, or?

Reply:
Uhh true :see_no_evil: